src/languageservice/services/yamlCompletion.ts

```diff
@@ -910,13 +910,8 @@ export class YamlCompletion {
                 if (index < s.schema.items.length) {
                   this.addSchemaValueCompletions(s.schema.items[index], separatorAfter, collector, types, 'value');
                 }
-              } else if (
-                typeof s.schema.items === 'object' &&
-                (s.schema.items.type === 'object' || isAnyOfAllOfOneOfType(s.schema.items))
-              ) {
-                this.addSchemaValueCompletions(s.schema.items, separatorAfter, collector, types, 'value', true);
               } else {
-                this.addSchemaValueCompletions(s.schema.items, separatorAfter, collector, types, 'value');
+                this.addSchemaValueCompletions(s.schema.items, separatorAfter, collector, types, 'value', true);
```

Comment:
unify the code for both scenarios. If a schema contains `items` it's an array. No need to check if `type === 'object'`